[Agentless Connectors] Integration overview panel

jedrazb opened the pull request:
## Summary

Add links to relevant areas for agentless connector:
- link integration owning this connector
- if policy related to this connector exist:
   -  add link to view policy
   - otherwise display `No policy found`
- if agent related to this policy exists 
  - display agent status (e.g. is it offline, or any other state)
  - link to fleet agent host overview
  - create discover app link to show logs related to agent/connector for the past 6 hrs (select relevant columns :) )
  - otherwise, handle this 

Changed isWaitingOnAgentlessDeployment to just check whether `last_seen` (connector heartbeat field) was populated, this is to address this reported [issue](https://github.com/elastic/search-team/issues/9165). 

### Happy path

https://github.com/user-attachments/assets/76db0b66-dda3-43a7-a74b-79348c7e7989

### Handling edge cases

#### No agent / inactive agent

https://github.com/user-attachments/assets/eae9b8da-de24-4d77-bc9e-b7cfad75b426

similarly we handle edge cases with policy not existing yet


### Checklist

Check the PR satisfies following conditions. 

Reviewers should verify this PR satisfies this list as well.

- [x] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/main/src/platform/packages/shared/kbn-i18n/README.md)
- [ ] [Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html) was added for features that require explanation or tutorials
- [ x [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios
- [ ] If a plugin configuration key changed, check if it needs to be allowlisted in the cloud and added to the [docker list](https://github.com/elastic/kibana/blob/main/src/dev/build/tasks/os_packages/docker_generator/resources/base/bin/kibana-docker)
- [x] This was checked for breaking HTTP API changes, and any breaking changes have been approved by the breaking-change committee. The `release_note:breaking` label should be applied in these situations.
- [ ] [Flaky Test Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was used on any tests changed
- [x] The PR  description includes the appropriate Release Notes section, and the correct `release_note:*` label is applied per the [guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)





<!--ONMERGE {"backportTargets":["9.0"]} ONMERGE-->

elasticmachine commented:
<!-- buildkite-pr-bot-trigger-comment -->:robot: Jobs for this PR can be triggered through checkboxes. :construction: 

:information_source: To trigger the CI, please tick the checkbox below :point_down: 

- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-pull-request --> Click to trigger **kibana-pull-request** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-project-from-pr --> Click to trigger **kibana-deploy-project-from-pr** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-cloud-from-pr --> Click to trigger **kibana-deploy-cloud-from-pr** for this PR!

jedrazb commented on x-pack/solutions/search/plugins/enterprise_search/public/applications/enterprise_search_content/components/connector_detail/connector_stats.tsx:
```diff
@@ -50,6 +67,92 @@ export interface StatCardProps {
   title: string;
 }
 
+function getStatusComponent({
```
Copied from private fleet logic, we might want to adapt to our styling later

jedrazb commented on x-pack/solutions/search/plugins/enterprise_search/public/applications/enterprise_search_content/components/connector_detail/connector_stats.tsx:
```diff
@@ -315,6 +448,149 @@ export const ConnectorStats: React.FC<ConnectorStatsProps> = ({ connector, index
+                      )}
+                      iconType="discoverApp"
+                      onClick={() => {
+                        discover.locator?.navigate({
```
I'm aware that we ended up with a lot of logic in this component ...

JoseLuisGJ replied:
Regarding this logic. If I'm not wrong we should avoid to keep using the KEA logic approach. The same concern was raised working with Crawlers in Serverless. I guess we should at least create a ticket to track this issue.

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/274839)
* Commit: 8c0520fab0c728f3b11ab0718b96c193df5082b1

### Failed CI Steps
* [Quick Checks](https://buildkite.com/elastic/kibana-pull-request/builds/274839#0194f0c5-118d-4d0f-b135-f8ac239db148)

### History
* :broken_heart: [Build #274824](https://buildkite.com/elastic/kibana-pull-request/builds/274824) failed 4cb13ac030970d6ee28b0cf5e4819fe54c8ceccf
* :broken_heart: [Build #274804](https://buildkite.com/elastic/kibana-pull-request/builds/274804) failed f86f8388a0a86eb0f43c2a5d1d6b8861e56a4925

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/274839","number":274839,"commit":"8c0520fab0c728f3b11ab0718b96c193df5082b1"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/274824","number":274824,"commit":"4cb13ac030970d6ee28b0cf5e4819fe54c8ceccf"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/274804","number":274804,"commit":"f86f8388a0a86eb0f43c2a5d1d6b8861e56a4925"}],"number":274839}
buildkite-pr-comment-->

elasticmachine commented:
## :broken_heart: Build Failed
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/274990)
* Commit: d0ecbaa433850b2b0a00386fe3491da96e51038b

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/274990#0194f40e-c15b-4fa6-b02b-5cae84a62374)

### History
* :broken_heart: [Build #274839](https://buildkite.com/elastic/kibana-pull-request/builds/274839) failed 8c0520fab0c728f3b11ab0718b96c193df5082b1
* :broken_heart: [Build #274824](https://buildkite.com/elastic/kibana-pull-request/builds/274824) failed 4cb13ac030970d6ee28b0cf5e4819fe54c8ceccf
* :broken_heart: [Build #274804](https://buildkite.com/elastic/kibana-pull-request/builds/274804) failed f86f8388a0a86eb0f43c2a5d1d6b8861e56a4925

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/274990","number":274990,"commit":"d0ecbaa433850b2b0a00386fe3491da96e51038b"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/274839","number":274839,"commit":"8c0520fab0c728f3b11ab0718b96c193df5082b1"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/274824","number":274824,"commit":"4cb13ac030970d6ee28b0cf5e4819fe54c8ceccf"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/274804","number":274804,"commit":"f86f8388a0a86eb0f43c2a5d1d6b8861e56a4925"}],"number":274990}
buildkite-pr-comment-->

JoseLuisGJ commented on x-pack/solutions/search/plugins/enterprise_search/public/applications/enterprise_search_content/components/connector_detail/connector_stats.tsx:
```diff
@@ -315,6 +390,114 @@ export const ConnectorStats: React.FC<ConnectorStatsProps> = ({ connector, index
+                      )}
+                    </EuiButtonEmpty>
+                  ) : (
+                    <EuiText size="s" color="warning">
+                      {noAgentLabel}
+                    </EuiText>
```
I think is redundant to display twice _No agent found_ I'd leave only the on in the badge. The footer should only contain actionable links
```suggestion
                    null
```

![CleanShot 2025-02-11 at 09 59 09@2x](https://github.com/user-attachments/assets/6029ce60-445e-4bc7-a7e5-e542e24ad7bd)

JoseLuisGJ reviewed (commented):
Nice work. I added some suggestions regarding how to display the links and the related info

jedrazb commented on x-pack/solutions/search/plugins/enterprise_search/public/applications/enterprise_search_content/components/connector_detail/connector_stats.tsx:
```diff
@@ -315,6 +390,114 @@ export const ConnectorStats: React.FC<ConnectorStatsProps> = ({ connector, index
+            footer={
+              <EuiFlexGroup justifyContent="spaceBetween" alignItems="center">
+                <EuiFlexItem grow={false}>
+                  <EuiButton
+                    data-test-subj="connectorStatsViewLogsButton"
+                    aria-label={i18n.translate(
+                      'xpack.enterpriseSearch.connectors.connectorStats.viewLogsButtonLabel',
+                      { defaultMessage: 'View logs' }
+                    )}
+                    disabled={!agentlessAgentExists}
+                    iconType="discoverApp"
+                    onClick={() => {
+                      discover.locator?.navigate(navigateToDiscoverPayload);
+                    }}
+                  >
+                    {i18n.translate(
+                      'xpack.enterpriseSearch.connectors.connectorStats.viewLogsButtonLabel',
+                      { defaultMessage: 'View logs' }
+                    )}
+                  </EuiButton>
```
sounds good, changed :)

jedrazb commented:
@JoseLuisGJ addressed your feedback
<img width="1277" alt="Screenshot 2025-02-11 at 11 08 37" src="https://github.com/user-attachments/assets/624ba83e-eeea-42b9-a2e0-ed29a5d2b827" />
<img width="1263" alt="Screenshot 2025-02-11 at 11 08 50" src="https://github.com/user-attachments/assets/01015cf9-bd7d-44e5-8ebf-b9c392207688" />

JoseLuisGJ reviewed (approved):
LGTM 👍

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/275028)
* Commit: 23c6e425a111a7b39a44253d81467f72413a634d

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/9635cfa5268c6250111f8eb5edcd693311a47193) | [after](https://github.com/elastic/kibana/commit/23c6e425a111a7b39a44253d81467f72413a634d) | diff |
| --- | --- | --- | --- |
| `enterpriseSearch` | 1389 | 1391 | +2 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/9635cfa5268c6250111f8eb5edcd693311a47193) | [after](https://github.com/elastic/kibana/commit/23c6e425a111a7b39a44253d81467f72413a634d) | diff |
| --- | --- | --- | --- |
| `enterpriseSearch` | 1.5MB | 1.5MB | +4.7KB |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/9635cfa5268c6250111f8eb5edcd693311a47193) | [after](https://github.com/elastic/kibana/commit/23c6e425a111a7b39a44253d81467f72413a634d) | diff |
| --- | --- | --- | --- |
| `enterpriseSearch` | 38.3KB | 38.6KB | +249.0B |



### History
* :broken_heart: [Build #274990](https://buildkite.com/elastic/kibana-pull-request/builds/274990) failed d0ecbaa433850b2b0a00386fe3491da96e51038b
* :broken_heart: [Build #274839](https://buildkite.com/elastic/kibana-pull-request/builds/274839) failed 8c0520fab0c728f3b11ab0718b96c193df5082b1
* :broken_heart: [Build #274824](https://buildkite.com/elastic/kibana-pull-request/builds/274824) failed 4cb13ac030970d6ee28b0cf5e4819fe54c8ceccf
* :broken_heart: [Build #274804](https://buildkite.com/elastic/kibana-pull-request/builds/274804) failed f86f8388a0a86eb0f43c2a5d1d6b8861e56a4925

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/275028","number":275028,"commit":"23c6e425a111a7b39a44253d81467f72413a634d"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/274990","number":274990,"commit":"d0ecbaa433850b2b0a00386fe3491da96e51038b"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/274839","number":274839,"commit":"8c0520fab0c728f3b11ab0718b96c193df5082b1"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/274824","number":274824,"commit":"4cb13ac030970d6ee28b0cf5e4819fe54c8ceccf"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/274804","number":274804,"commit":"f86f8388a0a86eb0f43c2a5d1d6b8861e56a4925"}],"number":275028}
buildkite-pr-comment-->

artem-shelkovnikov commented on x-pack/solutions/search/plugins/search_connectors/server/services/index.ts:
```diff
@@ -248,6 +262,48 @@ export class AgentlessConnectorsInfraService {
+        // If no agents assigned to policy, just return the policy
+        return policy;
+      } else {
+        // Return the first (and only) agentless host associated with this policy
```
Shall we check if there are several agents and warn if so?

jedrazb replied:
Addressed in https://github.com/elastic/kibana/pull/210222/commits/1897dacd79d7b8104a7a203ade9c09566ef0d4f8

artem-shelkovnikov commented on x-pack/solutions/search/plugins/search_connectors/server/services/index.ts:
```diff
@@ -25,10 +29,15 @@ export interface PackageConnectorSettings {
+  package_name: string;
   agent_policy_ids: string[];
   connector_settings: PackageConnectorSettings;
+  agent_metadata?: AgentMetadata;
```
We should document why it's not available in other methods.

Ideally it should be:
1. Always populated
2. A different object that outlines what is returned so that it's more obvious why and what is returned

If we keep adding nullable fields that are populated conditionally, it quickly becomes a large problem.

jedrazb replied:
you are right 👍 I will create a new tsx type for this method the returns agent metadata

jedrazb replied:
Addressed in https://github.com/elastic/kibana/pull/210222/commits/1897dacd79d7b8104a7a203ade9c09566ef0d4f8

artem-shelkovnikov reviewed (commented):
Looks good, left a couple comments

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/275106)
* Commit: 1897dacd79d7b8104a7a203ade9c09566ef0d4f8

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/9635cfa5268c6250111f8eb5edcd693311a47193) | [after](https://github.com/elastic/kibana/commit/1897dacd79d7b8104a7a203ade9c09566ef0d4f8) | diff |
| --- | --- | --- | --- |
| `enterpriseSearch` | 1389 | 1391 | +2 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/9635cfa5268c6250111f8eb5edcd693311a47193) | [after](https://github.com/elastic/kibana/commit/1897dacd79d7b8104a7a203ade9c09566ef0d4f8) | diff |
| --- | --- | --- | --- |
| `enterpriseSearch` | 1.5MB | 1.5MB | +4.7KB |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/9635cfa5268c6250111f8eb5edcd693311a47193) | [after](https://github.com/elastic/kibana/commit/1897dacd79d7b8104a7a203ade9c09566ef0d4f8) | diff |
| --- | --- | --- | --- |
| `enterpriseSearch` | 38.3KB | 38.6KB | +249.0B |



### History
* :green_heart: [Build #275028](https://buildkite.com/elastic/kibana-pull-request/builds/275028) succeeded 23c6e425a111a7b39a44253d81467f72413a634d
* :broken_heart: [Build #274990](https://buildkite.com/elastic/kibana-pull-request/builds/274990) failed d0ecbaa433850b2b0a00386fe3491da96e51038b
* :broken_heart: [Build #274839](https://buildkite.com/elastic/kibana-pull-request/builds/274839) failed 8c0520fab0c728f3b11ab0718b96c193df5082b1
* :broken_heart: [Build #274824](https://buildkite.com/elastic/kibana-pull-request/builds/274824) failed 4cb13ac030970d6ee28b0cf5e4819fe54c8ceccf
* :broken_heart: [Build #274804](https://buildkite.com/elastic/kibana-pull-request/builds/274804) failed f86f8388a0a86eb0f43c2a5d1d6b8861e56a4925

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/275106","number":275106,"commit":"1897dacd79d7b8104a7a203ade9c09566ef0d4f8"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/275028","number":275028,"commit":"23c6e425a111a7b39a44253d81467f72413a634d"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/274990","number":274990,"commit":"d0ecbaa433850b2b0a00386fe3491da96e51038b"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/274839","number":274839,"commit":"8c0520fab0c728f3b11ab0718b96c193df5082b1"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/274824","number":274824,"commit":"4cb13ac030970d6ee28b0cf5e4819fe54c8ceccf"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/274804","number":274804,"commit":"f86f8388a0a86eb0f43c2a5d1d6b8861e56a4925"}],"number":275106}
buildkite-pr-comment-->

kibanamachine commented:
Starting backport for target branches: 9.0

https://github.com/elastic/kibana/actions/runs/13266404741

kibanamachine commented:
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|9.0|[<img src="https://img.shields.io/github/pulls/detail/state/elastic/kibana/210609">](https://github.com/elastic/kibana/pull/210609)|

Note: Successful backport PRs will be merged automatically after passing CI.

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

<!--- Backport version: 9.4.3 -->
